crates/ethereum-forks/src/forkcondition.rs

```diff
@@ -85,9 +85,17 @@ impl ForkCondition {
     /// - The condition is satisfied by the timestamp;
     /// - or the condition is satisfied by the total difficulty
     pub fn active_at_head(&self, head: &Head) -> bool {
-        self.active_at_block(head.number) ||
-            self.active_at_timestamp(head.timestamp) ||
-            self.active_at_ttd(head.total_difficulty, head.difficulty)
+        match self {
+            Self::TTD { fork_block: Some(_), .. } => {
+                self.active_at_block(head.number) &&
+                    self.active_at_ttd(head.total_difficulty, head.difficulty)
```

Comment:
ah, we actually can't do `&&` here, because the `fork_block` corresponds to the `mergeNetsplitBlock` in forkid calculation, which is used in sepolia. It's essentially a block fork for the timestamp. And the block is _below_ the actual ttd merge block. But it's not used for any fork feature-activation.

see the docs here:
https://github.com/paradigmxyz/reth/blob/148ae1cc0a47116b19d6a837e63afa2a085d74ee/crates/ethereum-forks/src/forkcondition.rs#L12-L17

Reply:
Ok understand, I had a mis-interpretation, just fixed everything and tests, thanks